python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -323,6 +328,42 @@ def write(self, value: Any):
             )
             raise TypeError(msg) from e
 
+        # serialized_value.total_bytes *only* includes the size of the data. It does not
+        # include the size of the metadata, so we must account for the size of the
+        # metadata explicitly.
+        size = serialized_value.total_bytes + len(serialized_value.metadata)
+        if size > self._typ.buffer_size_bytes:
+            # Now make the channel backing store larger.
```

Comment:
Probably would be best to separate this out into a helper method.

Reply:
Done.